### PR TITLE
F1 2018

### PR DESCRIPTION
**Pull request by mrbelowski**
this is just the spotter implementation, and has been written without actually testing any of it. I'd like to get this into master (even if it's disabled) as I think the spotter could be worth having. If it turns out to be shit we can just disable the game type and remove the properties.

The spotter makes a major assumption about the player's yaw - if this is incorrect the co-ordinate system rotation will be wrong and the spotter calls will be nonsense.

The mapper is just a stub, but all the game's UDP packets are received and included in the raw gamestate data it receives

### CODE REVIEW

File: CrewChiefV4/CrewChiefV4.csproj [hunk 1]
```diff
@@ -388,6 +409,7 @@
     <Content Include="..\help.txt" />
     <Content Include="..\README.txt" />
   </ItemGroup>
+  <ItemGroup />
```
Comment (TheIronWolfModding) [inline CrewChiefV4/CrewChiefV4.csproj]: Empy ItemGroup?

File: CrewChiefV4/F1_2018/F12018Spotter.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+using CrewChiefV4.Events;
+using CrewChiefV4.Audio;
+using CrewChiefV4.GameState;
+using F1UdpNet;
+
+namespace CrewChiefV4.F1_2018
+{
+    class F12018Spotter : Spotter
+    {
+        private float twoPi = (float)(2 * Math.PI);
```
Comment (TheIronWolfModding) [inline CrewChiefV4/F1_2018/F12018Spotter.cs]: Probably worth either moving to Utilities class or introducing Constants class.

File: CrewChiefV4/ui_text.txt [hunk 3]
```diff
@@ -593,6 +609,11 @@ pcars64_launch_exe_help = This is the location of Steam.exe or the direct path t
 pcars64_launch_exe_filter = PCARS_64BIT
 pcars64_launch_exe_category = UI_STARTUP_AND_PATHS
 
+f1_2018_launch_exe = F1 2018 launch exe
+f1_2018_launch_exe_help = This is the location of Steam.exe or the direct path to f1_2018.exe. Use forward slashes to separate paths.
+f1_2018_launch_exe_filter = F1_2018
```
Comment (TheIronWolfModding) [inline CrewChiefV4/ui_text.txt]: Probably not worth doing now, but, something to keep in mind.

Currently, bunch of properties that don't work in F1 2018 will show up if game is selected.  The fix is to set filter of those to _filter = !F1_2018 (or, !F1_2018;IRACING for those that already excluded from IRACING)  Note to self: need to check if _filter = !IRACING still treats property as common for other games.

File: CrewChiefV4/F1_2018/F12018Spotter.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                    CarMotionData opponentData = currentWrapper.packetMotionData.m_carMotionData[i];
+                    float[] currentPositions = new float[] { opponentData.m_worldPositionX, opponentData.m_worldPositionZ };
+                    currentOpponentPositions.Add(currentPositions);
+                }
+                if (currentOpponentPositions.Count() > 0)
+                {
+                    float playerRotation = playerData.m_yaw;
+                    if (playerRotation < 0)
+                    {
+                        playerRotation = playerRotation * -1;
```
Comment (TheIronWolfModding) [inline CrewChiefV4/F1_2018/F12018Spotter.cs]: playerRotation = Math.Abs(playerRotation) ?  No need for if/else then

Comment (mrbelowski) [reply CrewChiefV4/F1_2018/F12018Spotter.cs]: don't think so, unless i'm missing something. The else block here subtracts the value from 2pi

File: CrewChiefV4/CrewChiefV4.csproj (hunk 1 above)
Comment (mrbelowski) [reply CrewChiefV4/CrewChiefV4.csproj]: weird

**Pull request merged**